internal/backend/db/dbgorm/values.go

```diff
@@ -0,0 +1,90 @@
+package dbgorm
+
+import (
+	"context"
+	"errors"
+
+	"google.golang.org/protobuf/proto"
+	"gorm.io/gorm"
+
+	"go.autokitteh.dev/autokitteh/internal/backend/db/dbgorm/scheme"
+	"go.autokitteh.dev/autokitteh/internal/kittehs"
+	"go.autokitteh.dev/autokitteh/sdk/sdkerrors"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+const maxValueSize = 64 * 1024
+
+func (db *gormdb) SetValue(ctx context.Context, envID sdktypes.EnvID, key string, v sdktypes.Value) error {
+	if !envID.IsValid() {
+		return sdkerrors.NewInvalidArgumentError("invalid env_id")
+	}
+
+	if v.ProtoSize() > maxValueSize {
+		return sdkerrors.NewInvalidArgumentError("value too large > %d bytes", maxValueSize)
+	}
+
+	uid, err := userIDFromContext(ctx)
+	if err != nil {
+		return err
+	}
+
+	return translateError(db.transaction(ctx, func(tx *tx) error {
+		db := tx.db
+
+		oo, err := tx.owner.EnsureUserAccessToEntitiesWithOwnership(tx.ctx, db, uid, envID.UUIDValue())
+		if err != nil {
+			if errors.Is(err, gorm.ErrRecordNotFound) {
```

Comment:
I think you don't need this. In Var it's implemented this var since scope could be Env or Connection, means we cannot set foreign key consrtaint. In value you could, so I'd prefer to have foreign key constrain which would be enforced in DB and not in logic

Reply:
removed